nengo/spa/spa.py

```diff
@@ -80,153 +76,20 @@ def __init__(self):
 
     def __init__(self, label=None, seed=None, add_to_container=None,
                  vocabs=None):
-        super(SPA, self).__init__(label, seed, add_to_container)
+        warnings.warn(DeprecationWarning(
+            "nengo.spa.SPA is deprecated. Use nengo.spa.Module instead."))
+
+        if seed is not None:
+            rng = np.random.RandomState(seed)
+        else:
+            rng = None
         vocabs = [] if vocabs is None else vocabs
```

Comment:
`SPA` expects `vocabs` to be a list whereas `Module` expects a `VocabularyMap`. Thus, a replacement of `SPA` with `Module` might require the `vocabs` argument to be adjusted (which is trivial and should not change behavior, but has to be done).

Technically, this line would not be required as `VocabularyMap` does the exact same thing if it gets passed a `None`. Creating the `VocabularyMap` here allows to use a common `super` call, otherwise branching would be required to do the call with a varying number of arguments. Passing in the `VocabularyMap` to the constructor means that a potential config of `vocabs` will be overwritten, but prior to this PR this config param did not exist and `SPA` had to be the root module and thus this can't break any existing code (but another reason to deprecate SPA to avoid potential problems from that slightly different behavior). Creating the `VocabularyMap` in this place requires an `rng` which explains the lines above the `vocabs = ` line.

Not sure how clear this answer was ... I feel like it's a lot of minute details why exactly I implemented it in this way.